app/uk/gov/hmrc/agentaccesscontrol/connectors/GovernmentGatewayProxyConnector.scala

```diff
@@ -23,19 +42,32 @@ class GovernmentGatewayProxyConnector(baseUrl: URL, httpPost: HttpPost) {
   }
 
   def parseResponse(xmlString: String) : Option[AgentDetails] = {
-    val XML = scala.xml.XML.withSAXParser(SAXParserFactory.newInstance().newSAXParser())
-
-    val xml = XML.loadString(xmlString)
+    val xml: Elem = toXmlElement(xmlString)
     val agentDetails = xml \ "AllocatedAgents" \ "AgentDetails"
-    val agentCode = (agentDetails \ "AgentCode").head.text
+    if (agentDetails.nonEmpty) {
+      val agentCode = (agentDetails \ "AgentCode").head.text
 
-    val credentials = (agentDetails \ "AssignedCredentials" \ "Credential").map { elem =>
-      AssignedCredentials((elem \ "CredentialIdentifier").head.text)
+      val credentials = (agentDetails \ "AssignedCredentials" \ "Credential").map { elem =>
+        AssignedCredentials((elem \ "CredentialIdentifier").head.text)
+      }
+      Some(AgentDetails(agentCode, credentials))
```

Comment:
Based on the top level element (`AllocatedAgents`) there can be more agents, which case is not handled here.

```
	<AllocatedAgents>
		<AgentDetails>
			<AgentCode>123456ABCDEF</AgentCode>
			<AssignedCredentials>
				<Credential>
					<CredentialIdentifier>001</CredentialIdentifier>
				</Credential>
			</AssignedCredentials>
		</AgentDetails>
		<AgentDetails>
			<AgentCode>123456789ABC</AgentCode>
			<AssignedCredentials>
				<Credential>
					<CredentialIdentifier>007</CredentialIdentifier>
				</Credential>
			</AssignedCredentials>
		</AgentDetails>
	</AllocatedAgents>
```

would return `Some(AgentDetails("123456ABCDEF", Seq("001", "007"))`, while agent 007 does not belong to that agency.